core/src/main/scala/org/apache/spark/storage/BlockManagerDecommissioner.scala

```diff
@@ -287,7 +287,8 @@ private[storage] class BlockManagerDecommissioner(
     val livePeerSet = bm.getPeers(false).toSet
     val currentPeerSet = migrationPeers.keys.toSet
     val deadPeers = currentPeerSet.diff(livePeerSet)
-    val newPeers = livePeerSet.diff(currentPeerSet)
+    // Randomize the orders of the peers to avoid migrating data to same nodes
```

Comment:
I think it's still possible to migrate to the same node but just different order, so maybe change the comment a bit as:
```suggestion
    // Randomize the orders of the peers to avoid hotspot nodes.
```